test/jdk/java/nio/channels/spi/SelectorProvider/inheritedChannel/StateTestService.java

```diff
@@ -38,26 +38,53 @@
  * message to indicate the test result.
  */
 import java.io.IOException;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.nio.ByteBuffer;
 import java.nio.channels.Channel;
 import java.nio.channels.DatagramChannel;
 import java.nio.channels.ServerSocketChannel;
 import java.nio.channels.SocketChannel;
+import java.nio.file.Path;
 
 public class StateTestService {
 
     static boolean failed = false;
     static int reply_port;
 
     static void check(boolean okay) {
+        println("check " + okay);
         if (!okay) {
             failed = true;
         }
     }
 
+    static String logDir;
+    static PrintStream p;
+    static boolean initialized = false;
+
+    // Opens named log file in ${test.classes}
+    static void initLogFile() {
+        if (initialized)
+            return;
+
+        try {
+            Path path = Path.of(logDir, "statetest.txt");
+            FileOutputStream f = new FileOutputStream(path.toFile(), true);
```

Comment:
This appends, is that intended? You could use Files.newOutputStream here to avoid the lossy conversion to File.

Reply:
Yes, there are multiple calls to StateTestService in the one test. So, they would be all overwritten by the last one. 

I had made the PR draft so I didn't realise there would be update notifications. But, thanks for the comments anyway. I will update and then re-open the review.